vmdb/app/models/mixins/container_provider_mixin.rb

```diff
@@ -0,0 +1,42 @@
+module ContainerProviderMixin
+  extend ActiveSupport::Concern
+
+  included do
+
+    has_many :container_nodes,                      :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_groups,                     :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_services,                   :foreign_key => :ems_id, :dependent => :destroy
+
+    def self.raw_api_endpoint(hostname, port)
+      URI::HTTPS.build(:host => hostname, :port => port.to_i)
+    end
+
+    # UI methods for determining availability of fields
+    def supports_port?
+      true
+    end
+
+    def api_endpoint
+      raw_api_endpoint(hostname, port)
+    end
+
+    def connect(_options = {})
+      self.class.raw_connect(hostname, port)
+    end
+
+    def authentication_check
+      # TODO: support real authentication using certificates
+      true
+    end
+
+    def verify_credentials(_auth_type = nil, _options = {})
+      # TODO: support real authentication using certificates
+      true
+    end
+
+    def authentication_status_ok?(_type = nil)
+      # TODO: support real authentication using certificates
+      true
+    end
```

Comment:
We should coordinate #2156.  I think we should merge that one first, and then this one should rebase against it.  Note that you might need verification of the OpenShift endpoint that's different from kubernetes, or perhaps checks both.